packages/protocol/contracts/dapis/Api3Adminship.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "./interfaces/IApi3Adminship.sol";
+
+/// @title Adminship contract controlled by the API3 DAO
+contract Api3Adminship is IApi3Adminship {
+  string internal constant ERROR_UNAUTHORIZED = "Unauthorized";
+  string internal constant ERROR_ZERO_ADDRESS = "Zero address";
+  string internal constant ERROR_EXPIRATION_NOT_EXTENDED = "Expiration not extended";
```

Comment:
What's the reason for storing the error messages to constants? Is it more gas cost efficient in case you use the same error message several times?

Reply:
The truth is that I copied them over from [here](https://github.com/api3dao/airnode/blob/871f17741045d4ff6c47dfb4f81f897e6beaf1f2/packages/protocol/contracts/authorizers/Api3Authorizer.sol#L8).

I don't really know if it costs less than using the same string error message several times but at least the [docs](https://docs.soliditylang.org/en/latest/contracts.html#constants) say that constants state cost is lower than regular state variables. 

> Compared to regular state variables, the gas costs of constant and immutable variables are much lower. For a constant variable, the expression assigned to it is copied to all the places where it is accessed and also re-evaluated each time. 

Maybe this is a question better suited for Burak and I also wonder if we should try to force their length to 32 bytes

Reply:
I'm currently asking some friends. The benefits of having smart contract auditor friends!

Reply:
<img width="503" alt="Screen Shot 2021-05-24 at 6 36 21 PM" src="https://user-images.githubusercontent.com/63020642/119415139-05fa5780-bcbf-11eb-8be0-6ceb061f3399.png">

Reply:
yea, looks like there isn't really a difference (only minor one during deploy). that's why i suggest removing the constant declarations and just putting string literals in the `require` statements. it makes the code simpler, since the error codes are quite simple to begin with.

Reply:
Inlining error strings is not an issue if they are only used once but if they are used in multiple places, you want to use variables so that you don't change one and forget to change the other. Once you use a variable for some you may as well use variables for all. But all in all it's just a pattern that I liked and picked up, it's not critical.